task_stages_integration_test.go

```diff
@@ -0,0 +1,113 @@
+package tfe
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTaskStagesRead(t *testing.T) {
+	skipIfBeta(t)
+
+	client := testClient(t)
+	ctx := context.Background()
+
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	defer orgTestCleanup()
+
+	runTaskTest, runTaskTestCleanup := createRunTask(t, client, orgTest)
+	defer runTaskTestCleanup()
+
+	wkspaceTest, wkspaceTestCleanup := createWorkspace(t, client, orgTest)
+	defer wkspaceTestCleanup()
+
+	wrTaskTest, wrTaskTestCleanup := createWorkspaceRunTask(t, client, wkspaceTest, runTaskTest)
+	defer wrTaskTestCleanup()
+
+	rTest, rTestCleanup := createRun(t, client, wkspaceTest)
+	defer rTestCleanup()
+
+	t.Run("without include param", func(t *testing.T) {
+		r, err := client.Runs.ReadWithOptions(ctx, rTest.ID, &RunReadOptions{
+			Include: "task_stages",
+		})
+		require.NoError(t, err)
+
+		taskStage, err := client.TaskStages.Read(ctx, r.TaskStages[0].ID, nil)
+		require.NoError(t, err)
+
+		assert.NotEmpty(t, taskStage.ID)
+		assert.NotEmpty(t, taskStage.Stage)
+		assert.NotNil(t, taskStage.StatusTimestamps.ErroredAt)
+		assert.NotNil(t, taskStage.StatusTimestamps.RunningAt)
+		assert.NotNil(t, taskStage.CreatedAt)
+		assert.NotNil(t, taskStage.UpdatedAt)
+		assert.NotNil(t, taskStage.Run)
+		assert.NotNil(t, taskStage.TaskResults)
+
+		// so this bit is interesting, if the relation is not specified in the include
+		// param, the fields of the struct will be zeroed out, minus the ID
+		assert.NotEmpty(t, taskStage.TaskResults[0].ID)
+		assert.Empty(t, taskStage.TaskResults[0].Status)
+		assert.Empty(t, taskStage.TaskResults[0].Message)
+	})
+
+	t.Run("with include param task_results", func(t *testing.T) {
+		r, err := client.Runs.ReadWithOptions(ctx, rTest.ID, &RunReadOptions{
+			Include: "task_stages",
+		})
+		require.NoError(t, err)
+
+		taskStage, err := client.TaskStages.Read(ctx, r.TaskStages[0].ID, &TaskStageReadOptions{
+			Include: "task_results",
+		})
+		require.NoError(t, err)
+
+		t.Run("task results are properly decoded", func(t *testing.T) {
+			assert.NotEmpty(t, taskStage.TaskResults[0].ID)
+			assert.NotEmpty(t, taskStage.TaskResults[0].Status)
+			assert.NotEmpty(t, taskStage.TaskResults[0].CreatedAt)
+			assert.Equal(t, wrTaskTest.ID, taskStage.TaskResults[0].WorkspaceTaskID)
+			assert.Equal(t, runTaskTest.Name, taskStage.TaskResults[0].TaskName)
+		})
+	})
+}
+
+func TestTaskStagesList(t *testing.T) {
+	skipIfBeta(t)
+
+	client := testClient(t)
+	ctx := context.Background()
+
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	defer orgTestCleanup()
```

Comment:
This is just an idle thought and not necessarily actionable, but I wonder if we ultimately need to make every cleanup call when the parent is to be cleaned up. I think all our relations are dependent: destroy, so it might be a waste of time.